src/pages/Sidebar/ChatList.tsx

```diff
@@ -0,0 +1,96 @@
+import { useForm } from 'react-hook-form';
+import {
+  ResizableHandle,
+  ResizablePanel,
+  ResizablePanelGroup,
+} from '@/components/ui/resizable';
+import { Input } from '@/components/ui/input';
+import { useGetChatsForUserQuery } from '@/redux/api/chatApi';
+import { useAppSelector } from '@/redux/hooks/reduxHooks';
+import { setSelectedChatId } from '@/redux/slices/selectionSlice.ts';
+import { useAppDispatch } from '@/redux/hooks/reduxHooks';
+import ChatItem from '@/pages/Sidebar/ChatItem.tsx';
+import NewChatDialog from '@/pages/Sidebar/NewChatDialog.tsx';
+import {
+  selectAuthUserId,
+  selectSelectedChatId,
+} from '@/redux/selectors/selectors';
+import { Participant } from '@/types/chatTypes.ts';
+
+const ChatList = () => {
+  const { register, watch } = useForm();
+  const authUserId = useAppSelector(selectAuthUserId);
+  const selectedChatId = useAppSelector(selectSelectedChatId);
+  const dispatch = useAppDispatch();
+
+  const {
+    data: chats = [],
+    isLoading,
+    isError,
+  } = useGetChatsForUserQuery(authUserId, {
```

Comment:
Ja preferuji využít [skiptoken](https://redux-toolkit.js.org/rtk-query/usage-with-typescript#skipping-queries-with-typescript-using-skiptoken). Přijde mi to rychlejší